web-ui/components/Staking/modals/stakingProcessModal.tsx

```diff
@@ -209,33 +203,58 @@ export const StakingProcessModal: React.FC<StakingModalProps> = ({ isOpen, onClo
   const { data: zone, isLoading: isZoneLoading } = useZoneQuery(selectedOption?.chainId ?? '');
 
   const valToByte = (val: number) => {
-    if (val > 1) {
-      val = 1;
-    }
-    if (val < 0) {
-      val = 0;
-    }
+    if (val > 1) val = 1;
+    if (val < 0) val = 0;
     return Math.abs(val * 200);
   };
 
+  // Function to add validator to memo
   const addValidator = (valAddr: string, weight: number) => {
-    let { words } = bech32.decode(valAddr);
-    let wordsUint8Array = new Uint8Array(bech32.fromWords(words));
-    let weightByte = valToByte(weight);
+    const { words } = bech32.decode(valAddr);
+    const wordsUint8Array = new Uint8Array(bech32.fromWords(words));
+    const weightByte = valToByte(weight);
+
     return Buffer.concat([Buffer.from([weightByte]), wordsUint8Array]);
   };
 
+  // Function to add mapped address to memo
+  const addMappedAccount = (mappedAddr: string) => {
+    if (!mappedAddr) {
+      return Buffer.alloc(0);
+    }
+    const { words } = bech32.decode(mappedAddr);
+    const wordsUint8Array = new Uint8Array(bech32.fromWords(words));
+
+    // Ensure the address is 20 bytes long (which is what 0x14 means)
+    if (wordsUint8Array.length !== 20) {
+      throw new Error('Mapped address must be 20 bytes long');
+    }
+
+    // 0x00 for field type, 0x14 for length, then the address bytes
+    return Buffer.concat([Buffer.from([0x00, 0x14]), wordsUint8Array]);
+  };
+
+  // Create the memo
   let memoBuffer = Buffer.alloc(0);
 
+  // Example address and intents
+
+  // Add mapped account to memo
+  memoBuffer = Buffer.concat([memoBuffer, addMappedAccount(address ?? 'quick1uwqjtgjhjctjc45ugy7ev5prprhehc7wje9uwh')]);
```

Comment:
Let's remove the default so you don't get everyone's qAssets :D

Reply:
lol not a good default huh... This was a hacky solution to ensure `addMappedAccounts` props are never undefined.

Reply:
So we only want to populate it for non-118 zones; and if a mapped account in not submitted and _is_ needed (non-118, and no previously used mapped account, then we send the tokens back to the origin account.